frontend/packages/integration-tests-cypress/tests/crud/namespace-crud.spec.ts

```diff
@@ -53,54 +83,122 @@ describe('Namespace', () => {
     nav.sidenav.clickNavLink(['Workloads', 'Secrets']);
-    projectDropdown.shouldContain('All Projects');
+    listPage.rows.shouldBeLoaded();
+    projectDropdown.shouldContain(allProjectsDropdownLabel);
     cy.log(
       'Nav back to list page from details page via breadcrumb should change Project back to "All Projects"',
     );
-    cy.get(`[data-test-rows="resource-row"]`)
-      .first()
-      .find('a')
-      .first()
+    listPage.filter.byName('alertmanager-main');
+    listPage.rows.clickRowByName('alertmanager-main');
+    detailsPage.isLoaded();
+    projectDropdown.shouldContain('openshift-monitoring');
+    detailsPage
+      .breadcrumb(0)
+      .contains('Secrets')
       .click();
-    projectDropdown.shouldNotContain('All Projects'); // after drilldown to Details page, project should be specific to resource
-    cy.byLegacyTestID('breadcrumb-link-0').click();
-    projectDropdown.shouldContain('All Projects');
+    listPage.rows.shouldBeLoaded();
+    projectDropdown.shouldContain(allProjectsDropdownLabel);
   });
 
   it('nav and breadcrumbs restores last selected Project when navigating from details to list view', () => {
     nav.sidenav.clickNavLink(['Workloads', 'Secrets']);
-    projectDropdown.selectProject('default');
-    projectDropdown.shouldContain('default');
-    cy.get(`[data-test-rows="resource-row"]`)
-      .first()
-      .find('a')
-      .first()
-      .click();
-    projectDropdown.shouldContain('default');
+    projectDropdown.selectProject(defaultProjectName);
+    projectDropdown.shouldContain(defaultProjectName);
+    listPage.rows.clickFirstLinkInFirstRow();
+    detailsPage.isLoaded();
+    projectDropdown.shouldContain(defaultProjectName);
     nav.sidenav.clickNavLink(['Workloads', 'Secrets']);
-    projectDropdown.shouldContain('default');
-    cy.get(`[data-test-rows="resource-row"]`)
-      .first()
-      .find('a')
-      .first()
+    projectDropdown.shouldContain(defaultProjectName);
+    listPage.rows.clickFirstLinkInFirstRow();
+    detailsPage.isLoaded();
+    projectDropdown.shouldContain(defaultProjectName);
+    detailsPage
+      .breadcrumb(0)
+      .contains('Secrets')
       .click();
-    projectDropdown.shouldContain('default');
-    cy.byLegacyTestID('breadcrumb-link-0').click();
-    projectDropdown.shouldContain('default');
+    listPage.rows.shouldBeLoaded();
+    projectDropdown.shouldContain(defaultProjectName);
+  });
+
+  ['Roles', 'RoleBindings'].forEach((rolesOrBindings) => {
+    it.only(`tests ${rolesOrBindings} and Cluster${rolesOrBindings} detail page breadcrumbs`, () => {
+      nav.sidenav.clickNavLink(['User Management', rolesOrBindings]);
+
+      cy.log(`test Cluster${rolesOrBindings} & All Projects`);
+      projectDropdown.selectProject(allProjectsDropdownLabel);
+      listPage.rows.shouldBeLoaded();
+      listPage.filter.by('cluster');
+      listPage.filter.numberOfActiveFiltersShouldBe(1);
+      listPage.filter.byName('alertmanager-main');
```

Comment:
This won't work on a non-OpenShift cluster since this ClusterRole won't exist. We should create a test project / test roles as needed.

Reply:
fixed, created example Role, RoleBinding, ClusterRole, and ClusterRoleBinding instances using test namespace/project.